test_pull_requests

```diff
@@ -499,192 +486,118 @@ end
-    def last_commit_for_pull_id(id, repo, num_tries=3)
+    def last_commit_for_pull_id(id, repo)
       commits = nil
       if $commits[repo]
         commits = $commits[repo][id]
       else
         $commits[repo] = {}
       end
       unless commits
-        (1..num_tries).each do |i|
-          res = get "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/pulls/%s/commits?per_page=100" % id
-          if res.success?
-            commits = res.data
-            $commits[repo][id] = commits
-            break
-          elsif i == num_tries
-            res.error!
-          end
-        end
+        auto_paginate = true
+        commits = pull_commits("#{@properties['github_user']}/#{repo}", id)
+        auto_paginate = false
       end
       commits.last
     end
 
-    def get_comments(issue_id, repo, num_tries=3)
-      page = 1
-      comments = []
-      last_comments = nil
-      while (!last_comments || last_comments.length == 100)
-        (1..num_tries).each do |i|
-          res = get "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/issues/%s/comments?page=#{page}&per_page=100" % issue_id
-          if res.success?
-            last_comments = res.data
-            comments += last_comments
-            page += 1
-            break
-          end
-          res.error! if i == num_tries
-        end
-      end
-      comments
+    def get_comments(issue_id, repo)
+      auto_paginate = true
+      pull_comments("#{@properties['github_user']}/#{repo}", issue_id)
+    ensure
+      auto_paginate = false
     end
 
-    def delete_comment(comment_id, repo, num_tries=3)
+    def delete_comment(comment_id, repo)
       $stderr.puts "  #{@dry_run_msg}Deleting comment ##{comment_id}"
-      (1..num_tries).each do |i|
-        res = delete "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/issues/comments/%s" % [comment_id]
-        # If we get 404 on a retry, that means there's no comment to delete
-        break if res.success? || (res.code == '404' && i > 1)
-        res.error! if i == num_tries
-      end if !@dry_run
+      delete_pull_comment("#{@properties['github_user']}/#{repo}", comment_id) if !@dry_run
     end
 
-    def add_comment(issue_id, repo, comment, num_tries=3)
+    def add_comment(issue_id, repo, comment)
       if @dry_run
-        $stderr.puts "  #{@dry_run_msg}add_comment"
+        $stderr.puts "  Dry run; skipping: add_comment"
         return {'id' => 0, 'body' => comment, 'updated_at' => Time.now}
-      end
-      params = { :body => comment }
-      (1..num_tries).each do |i|
-        res = post "#{GITHUB_API_BASE_URL}/repos/#{@properties['github_user']}/#{repo}/issues/%s/comments" % [issue_id], params
-        return res.data if res.success?
-        res.error! if i == num_tries
+      else
+        add_comment("#{@properties['github_user']}/#{repo}", issue_id, comment)
```

Comment:
so these are coming from the superclass? and we somehow don't overwrite this `add_comment()` from the OctoKit object with our own `add_comment()`? Yeek.